src/rgw/driver/sfs/sqlite/bindings/blob.h

```diff
@@ -13,28 +13,44 @@
  */
 #pragma once
 
+#include <tuple>
 #include <type_traits>
 
 #include "rgw/driver/sfs/sqlite/conversion_utils.h"
+// we need to include dbapi_type_wrapper.h only because including dbapi.h
+// creates circular dependencies
+#include "rgw/driver/sfs/sqlite/dbapi_type_wrapper.h"
 #include "rgw/driver/sfs/sqlite/sqlite_orm.h"
 #include "rgw_common.h"
 
-namespace sqlite_orm {
+namespace blob_utils {
 
-template <typename T>
-struct __is_sqlite_blob : std::false_type {};
+template <typename T, typename Tuple>
+struct has_type;
 
 template <typename T>
-inline constexpr bool is_sqlite_blob = __is_sqlite_blob<T>::value;
+struct has_type<T, std::tuple<>> : std::false_type {};
 
-template <>
-struct __is_sqlite_blob<rgw::sal::Attrs> : std::true_type {};
+template <typename T, typename U, typename... Ts>
+struct has_type<T, std::tuple<U, Ts...>> : has_type<T, std::tuple<Ts...>> {};
 
-template <>
-struct __is_sqlite_blob<ACLOwner> : std::true_type {};
+template <typename T, typename... Ts>
+struct has_type<T, std::tuple<T, Ts...>> : std::true_type {};
 
-template <>
-struct __is_sqlite_blob<rgw_placement_rule> : std::true_type {};
+// list of types that are stored as blobs and have the encode/decode functions
```

Comment:
This is nice! A comment like "To add automagic blob handling for a datastructure with ceph encode/decode add it here" would be awesome. I suspect we'll have to blobify something else eventually

Reply:
sure, I'll add a comment. 
In fact I was also thinking to do the same for this part: 
```c++
// by default type's decode function is under the ceph namespace
template <typename T>
struct __ceph_ns_decode : std::true_type {};

template <typename T>
inline constexpr bool ceph_ns_decode = __ceph_ns_decode<T>::value;

// specialize the ones that are not under the ceph namespace
template <>
struct __ceph_ns_decode<RGWAccessControlPolicy> : std::false_type {};
template <>
struct __ceph_ns_decode<RGWQuotaInfo> : std::false_type {};
template <>
struct __ceph_ns_decode<RGWObjectLock> : std::false_type {};
template <>
struct __ceph_ns_decode<RGWUserCaps> : std::false_type {};
template <>
struct __ceph_ns_decode<ACLOwner> : std::false_type {};
template <>
struct __ceph_ns_decode<rgw_placement_rule> : std::false_type {};
```
Adding a tuple and avoid to have to specialise this way. 
I'll add more verbose comments to explain how to add a new blob type